src/NuGet.Core/NuGet.Common/PathUtil/FileUtility.cs

```diff
@@ -92,7 +92,12 @@ public static void Replace(Action<string> writeSourceFile, string destFilePath)
                 throw new ArgumentNullException(nameof(destFilePath));
             }
 
-            var tempPath = GetTempFilePath(Path.GetDirectoryName(destFilePath));
+            string? destFileDirectory = Path.GetDirectoryName(destFilePath);
+            if (destFileDirectory is null)
```

Comment:
Should we also raise an exception if the destFileDirectory is an Empty string? [https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-7.0)

Reply:
It looks like an exception will be thrown either way (either by us, or by the BCL). The difference is that the `ArgumentException`'s `ParamName` value will be a different name, a name that doesn't exist in our API. But given our code is open source and this API only has a single path parameter, it's trivial to discover which parameter had an input error.

So, it's still the caller's responsibility to pass in sensible input. It doesn't cause the API to produce incorrect results. Since pushing changes resets approvals, unfortunately I don't feel motivated to add such a minor change (same goes for reverting my "see if this fixes E2E tests" commit)

Reply:
you could do a follow up PR :)